tests/catch/unit/stream/hipStreamSynchronize.cc

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANNTY OF ANY KIND, EXPRESS OR
+IMPLIED, INNCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANNY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER INN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR INN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <hip_test_common.hh>
+#include "streamCommon.hh"
+
+namespace hipStreamSynchronizeTest {
+
+/**
+ * @brief Check that hipStreamSynchronize handles empty streams properly.
+ *
+ */
+TEST_CASE("Unit_hipStreamSynchronize_EmptyStream") {
+  hipStream_t stream;
+  HIP_CHECK(hipStreamCreate(&stream));
+  HIP_CHECK(hipStreamSynchronize(stream));
+  HIP_CHECK(hipStreamDestroy(stream));
+}
+
+#if HT_AMD /* Disabled because frequency based wait is timing out on nvidia platforms */
+
+/**
+ * @brief Check that all work executing in a stream is finished after a call to
+ * hipStreamSynchronize.
+ *
+ */
+TEST_CASE("Unit_hipStreamSynchronize_FinishWork") {
+  const hipStream_t explicitStream = reinterpret_cast<hipStream_t>(-1);
+  hipStream_t stream = GENERATE_COPY(explicitStream, hip::nullStream, hip::streamPerThread);
+
+  if (stream == explicitStream) {
+    HIP_CHECK(hipStreamCreate(&stream));
+  }
+
+  HipTest::runKernelForDuration(std::chrono::milliseconds(500), stream);
+  HIP_CHECK(hipStreamSynchronize(stream));
+  HIP_CHECK(hipStreamQuery(stream));
+
+  if (stream != hip::nullStream && stream != hip::streamPerThread) {
```

Comment:
```suggestion
  const bool isExplicitStream = stream == explicitStream;
  if (isExplicitStream) {
    HIP_CHECK(hipStreamCreate(&stream));
  }

  HipTest::runKernelForDuration(std::chrono::milliseconds(500), stream);
  HIP_CHECK(hipStreamSynchronize(stream));
  HIP_CHECK(hipStreamQuery(stream));

  if (isExplicitStream) {
```

To make it more obvious that these are testing the same condition